libsigner/src/events.rs

```diff
@@ -534,9 +533,46 @@ fn process_new_burn_block_event<T: SignerEventTrait>(
         received_time: SystemTime::now(),
         burn_header_hash,
     };
-    if let Err(e) = request.respond(HttpResponse::empty(200u16)) {
-        error!("Failed to respond to request: {:?}", &e);
+    ack_dispatcher(request);
+    Ok(event)
+}
+
+/// Process a new burn block event from the node
+fn process_new_block<T: SignerEventTrait>(
+    mut request: HttpRequest,
+) -> Result<SignerEvent<T>, EventError> {
+    debug!("Got new_block event");
+    let mut body = String::new();
+    if let Err(e) = request.as_reader().read_to_string(&mut body) {
+        error!("Failed to read body: {:?}", &e);
+
+        ack_dispatcher(request);
+        return Err(EventError::MalformedRequest(format!(
+            "Failed to read body: {:?}",
+            &e
+        )));
+    }
+    #[derive(Debug, Deserialize)]
+    struct TempBlockEvent {
```

Comment:
Instead of making a bespoke and scoped struct, it might be cleaner to just add a `fn from_slice(bytes: &[u8]) -> Result<SignerEvent<T>, EventError>` function to `SignerEvent<T>`.  Then, you could put all of the decode logic into that function, and just pass in `body.as_bytes()`.

Reply:
I don't think I fully understand what you mean by passing in body.as_bytes(). But I did make a more generic process_event function that will correctly deserialize the request bytes so long as the event implements serde and try_from for SignerEvent<T>.